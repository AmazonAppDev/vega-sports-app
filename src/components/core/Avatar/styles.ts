import { StyleSheet } from 'react-native';

import type { AppTheme } from '@AppTheme';

export const getAvatarStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    wrapper: {
      alignItems: 'center',
    },
    imageWrapper: {
      backgroundColor: colors.surfaceContainer,
      borderColor: colors.transparent,
      borderWidth: 2,
      borderRadius: 20,
      alignItems: 'center',
      justifyContent: 'center',
      overflow: 'hidden',
    },
    imageWrapperFocused: {
      borderColor: colors.focusPrimary,
    },
    placeholder: {
      fontWeight: '800',
    },
    placeholderFocused: {
      color: colors.focusPrimary,
    },
    labelFocused: {
      color: colors.focusPrimary,
    },
  });

export const imageStyle = StyleSheet.create({
  image: { width: '100%', height: '100%' },
});
