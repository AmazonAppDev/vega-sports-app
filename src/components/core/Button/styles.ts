import { StyleSheet } from 'react-native';

export const getButtonStyles = () =>
  StyleSheet.create({
    button: {
      minWidth: 120,
    },
  });
