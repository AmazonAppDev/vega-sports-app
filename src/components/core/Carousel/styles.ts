import { StyleSheet } from 'react-native';

export const getCarouselStyles = () =>
  StyleSheet.create({
    wrapper: {
      alignItems: 'center',
    },
  });
