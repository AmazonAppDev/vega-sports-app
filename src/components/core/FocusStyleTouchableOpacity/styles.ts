import { StyleSheet } from 'react-native';

import type { AppTheme } from '@AppTheme';

export const getTouchableOpacityStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    base: {
      backgroundColor: colors.surfaceBright,
      borderColor: colors.transparent,
      borderRadius: 8,
      borderWidth: 4,
    },
    focused: {
      backgroundColor: colors.primaryFixed,
      borderColor: colors.primary,
      borderStyle: 'solid',
    },
  });
