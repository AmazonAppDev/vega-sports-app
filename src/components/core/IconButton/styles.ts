import { StyleSheet } from 'react-native';

import type { AppTheme } from '@AppTheme';

export const getIconButtonStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    disabledText: {
      opacity: 0.5,
    },
    button: {
      flexDirection: 'row',
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: colors.surfaceBright,
    },
    disabledButton: {
      backgroundColor: colors.surfaceDim,
    },
  });
