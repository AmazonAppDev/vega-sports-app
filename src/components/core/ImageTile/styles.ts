import { StyleSheet } from 'react-native';

import type { AppTheme } from '@AppTheme';

export const getImageTileStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    container: {
      flex: 1,
      alignContent: 'center',
      alignItems: 'center',
    },
    image: {
      backgroundColor: colors.onBackground,
      alignItems: 'center',
      justifyContent: 'flex-end',
    },
  });
