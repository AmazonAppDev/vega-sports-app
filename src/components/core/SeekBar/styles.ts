// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import { StyleSheet } from 'react-native';

import type { AppTheme } from '@AppTheme';
import { scaleUxToDp } from '@AppUtils';

export const getSeekBarStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    container: {
      position: 'absolute',
      alignItems: 'center',
      justifyContent: 'flex-end',
      height: '100%',
      width: '100%',
      backgroundColor: colors.transparent,
    },
    controls: {
      alignItems: 'flex-end',
      flexDirection: 'row',
      justifyContent: 'center',
      width: '90%',
      padding: scaleUxToDp(25),
      marginBottom: scaleUxToDp(50),
      zIndex: 2,
    },
    seekbar: {
      width: '90%',
      marginBottom: scaleUxToDp(10),
    },
    time: {
      color: colors.onPrimary,
      fontSize: scaleUxToDp(25),
      marginHorizontal: 30,
    },
    hiddenBar: {
      backgroundColor: colors.transparent,
      height: scaleUxToDp(10),
      width: '90%',
      position: 'absolute',
      top: 0,
      left: 0,
    },
    transparent: {
      backgroundColor: colors.transparent,
    },
  });

export const getSeekBarThumbIconStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    focused: {
      height: 32,
      width: 32,
      borderRadius: 18,
      justifyContent: 'center',
      alignItems: 'center',
      backgroundColor: colors.onPrimary,
    },
    unfocused: {
      height: 35,
      width: 10,
      backgroundColor: colors.background,
      borderWidth: 2,
      borderRadius: 5,
    },
  });

export const getForwardBackwardIconStyles = ({ colors }: AppTheme) =>
  StyleSheet.create({
    container: {
      alignItems: 'center',
      justifyContent: 'center',
      alignSelf: 'center',
      alignContent: 'center',
      flexDirection: 'row',
      height: 70,
      width: 150,
      marginBottom: 10,
    },
    multiplierText: {
      fontSize: 20,
      color: colors.onSecondary,
    },
    aboveThumb: {
      alignItems: 'center',
      justifyContent: 'center',
      height: 80,
      width: 120,
      display: 'flex',
      flexDirection: 'row',
    },
    fastForwardRewindLabel: {
      fontSize: 20,
      fontWeight: '700',
      lineHeight: 24,
      letterSpacing: 0.8,
      color: colors.onSecondary,
    },
    fastForwardRewindImage: {
      height: 24,
      width: 24,
    },
  });
