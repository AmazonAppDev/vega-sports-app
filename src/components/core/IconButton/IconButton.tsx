// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
// SPDX-License-Identifier: MIT-0

import React from 'react';

import MaterialCommunityIcon from '@amazon-devices/react-native-vector-icons/MaterialCommunityIcons';

import { useAppTheme, useThemedStyles } from '@AppTheme';
import {
  FocusStyleTouchableOpacity,
  type FocusStyleTouchableOpacityProps,
} from '@AppComponents/core/FocusStyleTouchableOpacity/FocusStyleTouchableOpacity';
import { getIconButtonStyles } from './styles';

type IconProps = React.ComponentProps<typeof MaterialCommunityIcon>;

export type IconButtonProps = {
  iconName: IconProps['name'];
  disabled?: boolean;
} & FocusStyleTouchableOpacityProps;

/**
 * Convenience icon button component that already applies proper styling to the underlying components
 */
export function IconButton({
  iconName,
  disabled = false,
  style,
  children,
  testID = 'icon-button',
  ...props
}: IconButtonProps) {
  const { typography } = useAppTheme();
  const styles = useThemedStyles(getIconButtonStyles);

  return (
    <FocusStyleTouchableOpacity
      disabled={disabled}
      style={[styles.button, disabled && styles.disabledButton, style]}
      role="button"
      testID={`${testID}-opacity`}
      {...props}>
      <MaterialCommunityIcon
        name={iconName}
        size={typography.size?.fontSize?.display?.lg}
        color={typography.text?.color?.display?.toString()}
        style={[disabled && styles.disabledText]}
        testID={`${testID}-icon`}
        aria-hidden
      />

      {children}
    </FocusStyleTouchableOpacity>
  );
}
