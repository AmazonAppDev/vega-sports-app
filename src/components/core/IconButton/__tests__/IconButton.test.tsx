import React from 'react';
import type { StyleProp, ViewStyle } from 'react-native';

import { Text } from '@AppComponents';
import { type AppTheme, lightTheme } from '@AppTheme';
import { renderWithTheme } from '@AppTestUtils/render';
import { IconButton } from '../IconButton';
import { getIconButtonStyles } from '../styles';

describe('IconButton', () => {
  const theme: AppTheme = {
    isDarkTheme: false,
    ...lightTheme,
  };

  const styles = getIconButtonStyles(theme);

  it('renders the button with the correct icon and styles', () => {
    const { getByTestId, toJSON } = renderWithTheme(
      <IconButton iconName="play" disabled={false} />,
    );

    const opacity = getByTestId('icon-button-opacity');
    const icon = getByTestId('icon-button-icon', {
      includeHiddenElements: true,
    });

    expect(opacity).toBeOnTheScreen();
    expect(opacity).toHaveStyle({
      ...styles.button,
    });

    expect(icon).toBeOnTheScreen();
    expect(icon).toHaveProp('name', 'play');
    expect(icon).toHaveStyle({});

    expect(toJSON()).toMatchSnapshot();
  });

  it('applies disabled styles when the button is disabled', () => {
    const { getByTestId, toJSON } = renderWithTheme(
      <IconButton iconName="pause" disabled={true} />,
      {
        theme,
      },
    );

    const opacity = getByTestId('icon-button-opacity');
    const icon = getByTestId('icon-button-icon', {
      includeHiddenElements: true,
    });

    expect(opacity).toBeOnTheScreen();
    expect(opacity).toHaveStyle({
      ...styles.button,
      ...styles.disabledButton,
    });

    expect(icon).toBeOnTheScreen();
    expect(icon).toHaveProp('name', 'pause');
    expect(icon).toHaveStyle({});

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders children when provided', () => {
    const text = 'Child Text';

    const { getByText, toJSON } = renderWithTheme(
      <IconButton iconName="stop" disabled={false}>
        <Text>{text}</Text>
      </IconButton>,
      {
        theme,
      },
    );

    const child = getByText(text);
    expect(child).toBeOnTheScreen();

    expect(toJSON()).toMatchSnapshot();
  });

  it('renders style prop with priority over internal styles', () => {
    const style: StyleProp<ViewStyle> = {
      paddingHorizontal: 80,
      backgroundColor: 'red',
    };

    const { getByTestId, toJSON } = renderWithTheme(
      <IconButton iconName="stop" disabled={true} style={style} />,
      {
        theme,
      },
    );

    const opacity = getByTestId('icon-button-opacity');
    const icon = getByTestId('icon-button-icon', {
      includeHiddenElements: true,
    });

    expect(opacity).toHaveStyle(style);
    expect(icon).toHaveProp('name', 'stop');
    expect(icon).toHaveStyle({});

    expect(toJSON()).toMatchSnapshot();
  });
});
