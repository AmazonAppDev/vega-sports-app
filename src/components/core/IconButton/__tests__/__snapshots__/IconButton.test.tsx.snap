// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconButton applies disabled styles when the button is disabled 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#ECEFF1",
      "borderColor": "transparent",
      "borderRadius": 8,
      "borderWidth": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="icon-button-opacity"
>
  <View
    aria-hidden={true}
    color="#121212"
    name="pause"
    size={37.5}
    style={
      [
        {
          "opacity": 0.5,
        },
      ]
    }
    testID="icon-button-icon"
  />
</View>
`;

exports[`IconButton renders children when provided 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "transparent",
      "borderRadius": 8,
      "borderWidth": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="icon-button-opacity"
>
  <View
    aria-hidden={true}
    color="#121212"
    name="stop"
    size={37.5}
    style={
      [
        false,
      ]
    }
    testID="icon-button-icon"
  />
  <Text
    accessibilityRole="text"
    role="presentation"
    style={
      {
        "color": "#121212",
        "fontFamily": "Amazon Ember",
        "fontSize": 20,
        "fontStyle": undefined,
        "fontWeight": "400",
        "letterSpacing": undefined,
        "lineHeight": 36,
      }
    }
  >
    Child Text
  </Text>
</View>
`;

exports[`IconButton renders style prop with priority over internal styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": true,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "red",
      "borderColor": "transparent",
      "borderRadius": 8,
      "borderWidth": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
      "paddingHorizontal": 80,
    }
  }
  testID="icon-button-opacity"
>
  <View
    aria-hidden={true}
    color="#121212"
    name="stop"
    size={37.5}
    style={
      [
        {
          "opacity": 0.5,
        },
      ]
    }
    testID="icon-button-icon"
  />
</View>
`;

exports[`IconButton renders the button with the correct icon and styles 1`] = `
<View
  accessibilityState={
    {
      "busy": undefined,
      "checked": undefined,
      "disabled": false,
      "expanded": undefined,
      "selected": undefined,
    }
  }
  accessibilityValue={
    {
      "max": undefined,
      "min": undefined,
      "now": undefined,
      "text": undefined,
    }
  }
  accessible={true}
  collapsable={false}
  focusable={false}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "borderColor": "transparent",
      "borderRadius": 8,
      "borderWidth": 4,
      "flexDirection": "row",
      "justifyContent": "center",
      "opacity": 1,
    }
  }
  testID="icon-button-opacity"
>
  <View
    aria-hidden={true}
    color="#121212"
    name="play"
    size={37.5}
    style={
      [
        false,
      ]
    }
    testID="icon-button-icon"
  />
</View>
`;
